src/main/java/org/zone/config/node/title/DefaultTitleFadeInNode.java

```diff
@@ -0,0 +1,113 @@
+        public String getDisplayId() {
+            return "zone.message.title.fadeIn.length";
+        }
+
+        @Override
+        public CommandArgument<Integer> getCommandArgument() {
+            return new IntegerArgument("length");
+        }
+
+        @Override
+        public CommandResult onChange(
+                CommandContext context, Integer newValue) {
+            try {
+                DefaultTitleFadeInNode.this.set(ZonePlugin.getZonesPlugin().getConfig(),
+                        new TimeDuration(DefaultTitleFadeInNode.this.getInitialValue().getTimeUnit(),
+                                newValue));
+                return CommandResult.success();
+            } catch (SerializationException se) {
+                se.printStackTrace();
+                return CommandResult.error(Messages.getZoneSavingError(se));
+            }
+        }
+    }
+
+    private class DefaultTitleFadeInConfigCommandTimeUnitsNode implements ConfigCommandNode<TimeUnits> {
+
+        @Override
+        public String getDisplayId() {
+            return "zone.message.title.fadeIn.timeUnit";
+        }
+
+        @Override
+        public CommandArgument<TimeUnits> getCommandArgument() {
+            return new EnumArgument<>("timeUnit", TimeUnits.class);
+        }
+
+        @Override
+        public CommandResult onChange(CommandContext context, TimeUnits newValue) {
+            try {
+                DefaultTitleFadeInNode.this.set(ZonePlugin.getZonesPlugin().getConfig(),
+                        new TimeDuration(newValue, 1));
+                return CommandResult.success();
+            } catch (SerializationException se) {
+                se.printStackTrace();
+                return CommandResult.error(Messages.getZoneSavingError(se));
+            }
+        }
+    }
+
+    @Override
+    public TimeDuration getDefault() {
+        return this.getInitialValue();
+    }
+
+    @Override
+    public Object[] getNode() {
+        return new Object[] {"zone", "message", "title", "fadeIn"};
+    }
+
+    @Override
+    public TimeDuration getInitialValue() {
+        return new TimeDuration(TimeUnits.SECONDS, 1);
+    }
+
+    @Override
+    public Collection<ConfigCommandNode<?>> getNodes() {
+        return Arrays.asList(new DefaultTitleFadeInConfigCommandTimeNode(), new DefaultTitleFadeInConfigCommandTimeUnitsNode());
+    }
+
+    @Override
+    public void set(CommentedConfigurationNode node, TimeDuration timeDuration)
+            throws SerializationException {
+        node.node("Length").set(timeDuration.getLength());
+        node.node("Unit").set(timeDuration.getTimeUnit().name());
+    }
+
+    @Override
+    public Optional<TimeDuration> get(CommentedConfigurationNode node) {
+        int length = node.node("Length").getInt();
+        String unitAsString = node.node("Unit").getString();
+        TimeUnits unit = TimeUnits.valueOf(unitAsString);
+        if (length < -1 || length == 0 || unitAsString == null) {
```

Comment:
apparently length ``<= 0`` isnt a thing.....

Reply:
must have dumped my brain in the garbage